Girder 5

zachmullen opened the pull request.

zachmullen committed:
Convert large_image plugin to Girder 5

zachmullen committed:
Convert large_image_annotation plugin to Girder 5

zachmullen committed:
Fix CI

zachmullen committed:
Add vue components to plugin namespace

zachmullen committed:
Expose setFrameQuad in module namespace

zachmullen committed:
Expose ElementCollection in public namespace

zachmullen committed:
Extend prototype when extending viewers

zachmullen committed:
Load geojs library in front-end before plugin code

zachmullen committed:
Don't double-load geojs

Also fixup image viewer extensions

zachmullen committed:
Undo removal of d3

zachmullen committed:
Load geojs after plugin JS in large_image plugin

zachmullen committed:
Upgrade d3 version to use modules

zachmullen committed:
Manually cache-bust geojs JS bundle

manthey committed:
Clean up old code by requiring somewhat new packages

Specifically, this expects a modern version of pyproj for the geospatial
packages.

zachmullen committed:
Remove remaining girder.logger references

zachmullen committed:
Add watch commands for front-end rebuilds

zachmullen committed:
Use more robust front-end file copy mechanism

This makes `npm run watch` work with the geojs copy step

zachmullen committed:
Make in-place extension of large_image views in the annotation plugin work as expected

zachmullen committed:
Remove asynchronous job flags

manthey commented:
There are some jobs that are run locally (so that they share the python environment of the girder server), and some that are run remotely (so that they use the environment of girder_worker).  This has been a virtue for, example, converting images to a more efficient format without needing to have a girder_worker node spun up.  If we convert all of these jobs to celery jobs, we might want to retain a deployment mode with a local queue that is distinct from the girder_worker queue.

zachmullen commented:
In Girder 5, anything that needs to be done asynchronously will be offloaded from the Girder HTTP server, with celery being the built-in choice for doing so. To accommodate that change, the idea is to make the celery environment as similar as possible to the server environment, chiefly by giving it access to the mongo database.

> we might want to retain a deployment mode with a local queue that is distinct from the girder_worker queue.

Indeed, this is the goal; I've started replacing these tasks with raw celery tasks rather than girder_worker tasks. The deployment mode now uses multiple processes and eschews background threads. See https://github.com/girder/histomics-dev-env/blob/main/docker-compose.yml for an example of a dev environment that reflects this.

zachmullen committed:
Port zarr dep changes to girder-5 branch

zachmullen committed:
Also add imagecodecs to zarr source reqs

naglepuff committed:
Merge branch 'master' into update-with-master

naglepuff committed:
Use girder 5 style logging

naglepuff committed:
Merge branch 'master' into update-with-master

naglepuff committed:
Merge branch 'master' into update-with-master

naglepuff committed:
Merge pull request #1712 from girder/update-with-master

Bring recent changes to `large_image` into the `girder5` branch

naglepuff committed:
Update tox config for girder 5

naglepuff committed:
Remove girder 3 style logging

naglepuff committed:
Remove web client test from python test suite

naglepuff committed:
Import WorkerSettings properly

naglepuff committed:
Use test.Dockerfile image for large image tests

Also use node 16, since that is what is used by the current girder_test
image.

naglepuff committed:
Pin zarr

naglepuff committed:
Update eslint config

Ensure that it works with the `girder` global and ignores the `dist/` directory.

naglepuff committed:
Format client directories

naglepuff committed:
Ignore G002 and G004 errors

Adding this primarily as a stopgap. It might be worthwhile to go back and clean this issues up before merging the Girder 5 branch.

naglepuff committed:
Format python

naglepuff committed:
Add format-only commits to blame ignore revs

naglepuff committed:
Use node 20 for tests

naglepuff committed:
Merge pull request #1728 from girder/girder-5-tox

Make CI testing compatible with large image and girder 5

naglepuff committed:
Build pre-release version from girder-5 branch

naglepuff committed:
Merge pull request #1745 from girder/girder-5-installable

Make pre-release versions of large_image for Girder 5

naglepuff committed:
Ignore type error for now

naglepuff committed:
Build large_image and annotation plugins on release

naglepuff committed:
Use tox-and-node image for release check job

naglepuff committed:
Check node versions in check_release

naglepuff committed:
Use correct node version

naglepuff committed:
Use nvm in check_release environment

naglepuff committed:
Merge branch 'master' into girder-5

naglepuff committed:
Add missing comma

naglepuff committed:
Remove unused type: ignore

naglepuff committed:
Disable dicom plugin tests

naglepuff committed:
Pin numpy to fix type tests

naglepuff committed:
Merge pull request #1779 from girder/girder-5-cleanup-tests

Remove unused type: ignore

naglepuff committed:
Merge branch 'master' into girder-5-unpin-numpy

naglepuff committed:
Unpin numpy

naglepuff committed:
Merge pull request #1782 from girder/girder-5-unpin-numpy

Unpin numpy in girder 5

naglepuff committed:
Specify toxandnode executor

naglepuff committed:
Make sure to source bashrc for nvm scripts to work

naglepuff committed:
Include built client code in distribution

naglepuff committed:
Use environment with nvm/node for release

This is necessary because we want to build the client packages for
girder plugins whose source code is contained within this repository.

manthey committed:
Combine the wheels and check release jobs

Publish wheel artifacts as part of the check release job.  Publish all
wheels using a single twine command so partial failures are better
handled.

naglepuff committed:
Merge pull request #1777 from girder/girder-5-build-client-on-release

Build plugin client code on release

manthey committed:
Fix building docker images

manthey committed:
Merge pull request #1786 from girder/single-twine

Combine the wheels and check-release CI jobs

naglepuff committed:
Merge branch 'master' into girder-5-update-20253101

naglepuff committed:
Use the `girder` constant to get `restRequest`

naglepuff committed:
Adjust import paths to fit new structure

naglepuff committed:
Initial girder 5 conversion

naglepuff committed:
Build client code for dicom plugin on release

naglepuff committed:
Merge pull request #1799 from girder/girder-5-update-20253101

Merge recent changes from `master` into `girder-5`

naglepuff committed:
Use `export default` over `module.exports`

naglepuff committed:
Ignore lint errors for now

naglepuff committed:
Add missing arg for registerPluginStaticContent

naglepuff committed:
Re-enable dicom tests

naglepuff committed:
Register dicomweb plugin namespace

zachmullen committed:
Fix JS bug in metadata editor widget

zachmullen committed:
Merge pull request #1817 from girder/fix-metadata-editor-widget

Fix JS bug in metadata editor widget

zachmullen committed:
Fix metadata widget bug in addMetadataByKey

zachmullen committed:
Merge pull request #1820 from girder/fix-metadata-editor-widget

Fix metadata widget bug in addMetadataByKey

naglepuff committed:
WIP run dicom tests

naglepuff committed:
Import widgets

naglepuff committed:
Merge pull request #1822 from girder/girder5-fix-widget-imports

Fix widget imports (Girder 5)

naglepuff committed:
Require girder 5

naglepuff committed:
Fix watch script

naglepuff committed:
Export views

naglepuff committed:
Fix imports

naglepuff committed:
Remove old girder 3 style client tests

naglepuff committed:
Fix path for DICOMweb assetstore plugin

naglepuff committed:
Remove dependency on phantomjs

naglepuff committed:
Merge pull request #1800 from girder/convert-dicomweb-plugin-to-girder5

Convert Dicom Web Assetstore Plugin to Girder 5

manthey committed:
Guard the assetstoreImportViewMap update

We can only set the assetstoreImportViewMap if the main Girder
application exists.

manthey committed:
Merge pull request #1845 from girder/guard-asseststore-map

Guard the assetstoreImportViewMap update

manthey committed:
Merge branch 'master' into merge-master-250310

manthey committed:
Run playwright tests in CI

manthey committed:
Merge pull request #1848 from girder/merge-master-250310

Merge master 250310

zachmullen committed:
Merge pull request #1849 from girder/playwright-ci

Run playwright tests in CI

zachmullen committed:
Re-enable tox tests

zachmullen committed:
Merge pull request #1850 from girder/run-pytest-tests

Run pytest tests

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Use environment variables for worker tests in girder 5.

manthey committed:
Merge pull request #1865 from girder/worker-settings

Use environment variables for worker tests in girder 5.

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Better handle long rest requests

For those that shouldn't be running as a local job, ensure that they
emit data in a timely fashion.

manthey committed:
Merge pull request #1871 from girder/long-requests

Better handle long rest requests in Girder 5

manthey committed:
Use a local task to ingest annotations

Adjust the function so it can be reused by HistomicsUI

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Merge pull request #1876 from girder/ingest-annotations

Use a local task to ingest annotations

manthey committed:
Make a change to make process_annotations easier for HUI to call

manthey committed:
Properly expose task to celery

manthey committed:
Merge pull request #1877 from girder/process-annotations

Properly expose annotation processing task to celery

manthey committed:
Generalize itemFromEvent

This allows it to be reused by HUI

manthey committed:
Merge branch 'master' into generalize-item-from-event

manthey committed:
Merge pull request #1880 from girder/generalize-item-from-event

Generalize item from event

manthey committed:
Improve task name

manthey committed:
Don't schedule a needless annotation handling task

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Remove explicit cache busting as girder now handles it

manthey committed:
Merge pull request #1901 from girder/remove-geojs-cache-busting

Remove explicit cache busting as girder now handles it

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Fix girder-5 patch elements endpoint

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Merge branch 'master' into girder-5

zachmullen committed:
Conform to new notifications API from Girder core

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Merge branch 'more-documentdb-workarounds' into girder-5

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Merge branch 'girder-5' into updated-notifications-api

manthey committed:
Update girder version needs for girder-5 branch

manthey committed:
Merge pull request #1961 from girder/girder-version

Update girder version needs for girder-5 branch

zachmullen committed:
Merge pull request #1938 from girder/updated-notifications-api

Conform to new notifications API from Girder core

manthey committed:
Hook events in girder_worker

manthey committed:
Merge pull request #1962 from girder/gw_plugin

Hook events in girder_worker

manthey committed:
Refactor how girder objects are tracked for annotations

Avoid using a local cache as it might get processed by another process

manthey committed:
Merge pull request #1963 from girder/refactor-annotation-reprocess

Refactor how girder objects are tracked for annotations

manthey committed:
Update girder-5 version

Handle changes in testing

manthey committed:
Merge pull request #1964 from girder/update-girder-5

Update girder-5 version

naglepuff committed:
Merge branch 'master' into girder-5

manthey committed:
Remove event stream polling fallback

manthey committed:
Merge pull request #1974 from girder/remove-eventstream-fallback

Remove event stream polling fallback

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Satisfy client linter

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Fix node version in python 3.14 test

manthey committed:
Update package-lock

manthey committed:
Merge pull request #2024 from girder/girder-5-update-package-lock

Update package-lock

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Expose channel colors so other plugins can use them.

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Update girder 5 testing to node 22

Node 20 is EOL

manthey committed:
Add a convenience script to rebuild package-locks.

manthey committed:
Merge pull request #2083 from girder/node-version

Update girder 5 testing to node 22

manthey committed:
Make some testing defaults more automatic

manthey committed:
Merge pull request #2084 from girder/update-script

Add a convenience script to rebuild package-locks.

manthey committed:
Update slim.Dockerfile

manthey committed:
Update girder-5 slim.Dockerfile

manthey committed:
Merge pull request #2085 from girder/update-slim-dockerfile

Update slim dockerfile

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Update test for Girder #3847

manthey committed:
Build girder client plugins as part of tox

manthey committed:
Merge pull request #2097 from girder/tox-build-client

Build girder client plugins as part of tox

manthey committed:
Clean up a girder-5 requirement

manthey committed:
Merge pull request #2099 from girder/girder-5-req

Clean up a girder-5 requirement

manthey committed:
Merge branch 'master' into girder-5

manthey committed:
Update packages

manthey committed:
Merge branch 'master' into girder-5
